controllers/v1/programOperationsController.js

```diff
@@ -691,6 +691,12 @@ module.exports = class ProgramOperations {
         return new Promise(async (resolve, reject) => {
             try {
 
+                if (!/^[a-zA-Z0-9_-]+$/.test(req.query.id)) {
```

Comment:
why is this added ? @praveenKDass

Reply:
This is a unfixed security issue added for sanitise the qurey.id. For Regular Expression Denial of Service (ReDoS)
issue